fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/Config.java

```diff
@@ -209,78 +120,184 @@ private Config(final long randomSeed, @Nonnull final Metric metric, final int nu
         this.maxNumConcurrentNeighborhoodFetches = maxNumConcurrentNeighborhoodFetches;
     }
 
-    public long getRandomSeed() {
-        return randomSeed;
+    /**
+     * Indicator that if {@code true} causes the insert logic of the HNSW to be seeded using a hash of the primary key
+     * of the record that is inserted. That can be useful for testing. If {@code isDeterministicSeeding} is
+     * {@code false}, we use {@link System#nanoTime()} for seeding.
```

Comment:
Is there a reason we need to seed at all if `deterministicSeeding` is `false`? Like, is there a reason not to use something like `ThreadLocalRandom`?